File: subsets-ii/subsets-ii.cpp
class Solution {
public:
    vector<vector<int>>m;
    set<vector<int>>s;
    void solve(vector<int>nums,vector<int>v)
    {
        if(nums.size()==0)
        {
            
            if(s.find(v)==s.end())
            {
                sort(v.begin(),v.end());
                s.insert(v);
                m.push_back(v);
                
            }
            return ;
        }
        vector<int>v1=v,v2=v;
        v1.push_back(nums[0]);
        nums.erase(nums.begin()+0);
        solve(nums,v1);
        solve(nums,v2);
        
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<int>v;
        sort(nums.begin(),nums.end());
        solve(nums,v);
        return m;
        
    }
};

File: 90-subsets-ii/subsets-ii.py
class Solution:
   def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
      
       nums.sort()
       ans = []
       arr = []
       def solve(ind:int, is_prev_selected:bool) -> None:
           if(ind == len(nums)):
               ans.append(tuple(arr))
               return None
           if ind>0 and nums[ind] == nums[ind-1] and not is_prev_selected:
               solve(ind+1,False)
           else:
               arr.append(nums[ind])
               solve(ind+1,True)
               arr.pop()
               solve(ind+1,False)
       solve(0,False)
       return ans